src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -113,6 +113,9 @@
     // A set of methods that result in dialog creations.
     protected static final Set<String> dialogCreatingMethods = new HashSet<String>();
 
+    private transient Set<SIPDialogInjectionListener> dialogInjectionListeners = 
```

Comment:
is this collection meant to be read/written concurrently?
Do we need a concurrent collection?
Is the listener adding order important? The current logic takes the first matching dialog, so im not sure order will be important (in that case better to use List)